services/workflows-service/src/common/interceptors/remove-temp-file.interceptor.ts

```diff
@@ -0,0 +1,53 @@
+import { AppLoggerService } from '@/common/app-logger/app-logger.service';
+import { Injectable, NestInterceptor, ExecutionContext, CallHandler } from '@nestjs/common';
+import { Observable, catchError, firstValueFrom, of, tap } from 'rxjs';
+import * as fs from 'fs';
+
+@Injectable()
+export class RemoveTempFileInterceptor implements NestInterceptor {
+  constructor(private readonly logger: AppLoggerService) {}
+  private async deleteTempFile(req: any) {
+    if (req?.file?.path) {
```

Comment:
the file in the request is no the file on your local system.
also, this is an overcomplicated n and should be just in the process of if you downloaded to your local env afield via the import-export process.  ensure the downloaded file is deleted.
and it should be a part of the process and not a part of the server's functionality/middleware.

Reply:
but i dont download any file, once ive got the file in the controller the file already uploaded to the filesystem

Reply:
I agree with you when in case our multer was `memoryStorage();`
since we are using file storage, i have to remove the file because multer already keep it on file system, that's why you can see that im skip the downloading part .

these days you dont *actually remove* them when running locally with file provider